zendesk/brand.go

```diff
@@ -0,0 +1,51 @@
+package zendesk
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// Brand is struct for brand payload
+// https://developer.zendesk.com/rest_api/docs/support/brands
+type Brand struct {
+	ID                int64      `json:"id,omitempty"`
+	URL               string     `json:"url,omitempty"`
+	Name              string     `json:"name"`
+	BrandURL          string     `json:"brand_url,omitempty"`
+	HasHelpCenter     bool       `json:"has_help_center,omitempty"`
+	HelpCenterState   string     `json:"help_center_state,omitempty"`
+	Active            bool       `json:"active,omitempty"`
+	Default           bool       `json:"default,omitempty"`
+	Logo              Attachment `json:"logo,omitempty"`
+	TicketFieldIDs    []int64    `json:"ticket_field_ids,omitempty"`
+	Subdomain         string     `json:"subdomain"`
+	HostMapping       string     `json:"host_mapping,omitempty"`
+	SignatureTemplate string     `json:"signature_template"`
+	CreatedAt         time.Time  `json:"created_at,omitempty"`
+	UpdatedAt         time.Time  `json:"updated_at,omitempty"`
+}
+
+// BrandAPI an interface containing all methods associated with zendesk brands
+type BrandAPI interface {
+	CreateBrand(brand Brand) (Brand, error)
+}
+
+// CreateBrand creates new brand
+// https://developer.zendesk.com/rest_api/docs/support/brands#create-brand
+func (z *Client) CreateBrand(brand Brand) (Brand, error) {
```

Comment:
this is fine but there are a lot of places in here where this receiver is not a pointer.

We might need to create an issue to update all of those.

Reply:
Thanks. I opened new issue about this #80